fmpz/comb.c

```diff
@@ -0,0 +1,351 @@
+/*
+    Copyright (C) 2008, 2009, William Hart 
+    Copyright (C) 2010 Fredrik Johansson
+    Copyright (C) 2021 Daniel Schultz
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include <gmp.h>
+#include "flint.h"
+#include "ulong_extras.h"
+#include "fmpz.h"
+#include "nmod_vec.h"
+#include "fmpz_vec.h"
+#include "fmpz_poly.h"
+
+/* The better mpn_addmul_1 is, the larger FMPZ_CRT_UI_CUTOFF can be. */
+#define FMPZ_CRT_UI_CUTOFF 50
+
+/*
+    The better fmpz_fdiv_ui (mpn_mod_1) is, the larger FMPZ_MOD_UI_CUTOFF can
+    be. Very poor implementations of mpn_mod_1 can have 10, and anything up to
+    100 is reasonable.
+*/
+#define FMPZ_MOD_UI_CUTOFF 75
+
+/* minimal number of basecase chunks in which to partition */
+#define FMPZ_CRT_UI_MULTIPLE_CUTOFF 8
+#define FMPZ_MOD_UI_MULTIPLE_CUTOFF 4
+
+
+void fmpz_comb_temp_init(fmpz_comb_temp_t CT, const fmpz_comb_t C)
```

Comment:
I really prefer these functions all in separate files. It makes it hard to find stuff if they are grouped in the same file. The only time other functions are permitted is if they are internal versions of the same thing (e.g. underscore versions) or if they are helper functions.

Reply:
ok. fine. But the old fmpz/comb_init.c had both `fmpz_comb_temp_init` and `fmpz_comb_init`.

Reply:
That is fine. I really mean have the init and clear functions in separate files.